### PR TITLE
Update data-fetching.md

### PR DESCRIPTION
**Pull request by jrioscloud**
Improve typing for use with Typescript example

<!--
Thanks for opening a PR! Your contribution is much appreciated.
In order to make sure your PR is handled as smoothly as possible we request that you follow the checklist sections below.
Choose the right checklist for the change that you're making:

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [X] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [X] Make sure the linting passes

### CODE REVIEW

File: docs/basic-features/data-fetching.md [hunk 1]
```diff
@@ -185,9 +185,9 @@ You should use `getStaticProps` if:
 For TypeScript, you can use the `GetStaticProps` type from `next`:
 
 ```ts
-import { GetStaticProps } from 'next'
+import { GetStaticProps, GetServerSidePropsContext } from 'next'
 
-export const getStaticProps: GetStaticProps = async (context) => {
+export const getStaticProps: GetStaticProps = async (context: GetServerSidePropsContext) => {
```
Comment (timneutkens) [inline docs/basic-features/data-fetching.md:190]: You're using `GetServerSidePropsContext` on `getStaticProps` 🤔

Comment (papaponmx) [reply docs/basic-features/data-fetching.md]: You are right, I've updated it 🤓

**Review by timneutkens — DISMISSED** [review]

**Review by styfle — DISMISSED** [review]

File: docs/basic-features/data-fetching.md [hunk 2]
```diff
@@ -185,9 +185,9 @@ You should use `getStaticProps` if:
 For TypeScript, you can use the `GetStaticProps` type from `next`:
 
 ```ts
-import { GetStaticProps } from 'next'
+import { GetStaticProps, GetStaticPropsContext } from 'next'
 
-export const getStaticProps: GetStaticProps = async (context) => {
+export const getStaticProps: GetStaticProps = async (context: GetStaticPropsContext) => {
```
Comment (leerob) [inline docs/basic-features/data-fetching.md:190]: Looks like there's a linting error

**Review by ijjk — APPROVED** [review]

**Comment by pcjmfranken** [comment]
The type `GetStaticProps` already implements this type constraint for its `context` argument. Redundantly adding it again makes no sense.

Not a big deal but this docs change should probably be reverted.

The `GetStaticProps` type, argument type included by default:

```ts
export type GetStaticProps<
  P extends { [key: string]: any } = { [key: string]: any },
  Q extends ParsedUrlQuery = ParsedUrlQuery
> = (
  context: GetStaticPropsContext<Q>
) => Promise<GetStaticPropsResult<P>> | GetStaticPropsResult<P>
```

Intellisense also works fine following the previous doc's implementation:
image

**Comment by styfle** [comment]
Good point! Reverted in #27148

**Pull request merged**